lib/msal-common/src/cache/CacheHelpers.ts

```diff
@@ -165,9 +165,17 @@ export class CacheHelpers {
         });
     }
 
+    /**
+     * Checks that any parameters are exact matches for key value, since key.match in the above functions only do contains checks, not exact matches.
+     * @param atKey 
+     * @param clientId 
+     * @param authority 
+     * @param resource 
+     * @param homeAccountIdentifier 
+     */
     private checkForExactKeyMatch(atKey: AccessTokenKey, clientId: string, authority: string, resource?: string, homeAccountIdentifier?: string): boolean {
         const hasClientId = (atKey.clientId === clientId);
-        const hasAuthorityUri = (atKey.authority === authority);
+        const hasAuthorityUri = !StringUtils.isEmpty(authority) ? (atKey.authority === authority) : true;
```

Comment:
This needs a comment to explain why this should be true if `authority` is an empty string.

Reply:
```suggestion
        const hasAuthorityUri = StringUtils.isEmpty(authority) || (atKey.authority === authority);
```

Reply:
This is checking for an exact key match, if there is no authority passed then we don't want to fail the check as we will get false negatives in our cache lookups. I will add a comment.

Reply:
Okay, maybe renaming the variable might help also, since its not truthful in that scenario.

Reply:
Added a comment and your suggestion, I can't think of a good var name though, so left it unchanged. feel free to suggest something. I changed this for all lines, so that's why i didn't commit  your suggestion.